### PR TITLE
AndroidX migration OSS test

### PR DESCRIPTION
**Pull request by mdvacca**
Summary: AndroidX migration OSS test

Differential Revision: D14200097

### CODE REVIEW

**Comment by dulmandakh** [comment]
Now tests failed not because of **is_androidx**, but due to class/package not found error. So we need to land AndroidX BUCK, and use it to rewrite dependencies in bzl if **is_androidx** specified.

**Comment by dulmandakh** [comment]
I created a PR to land AndroidX Buck, but I need some information about internal AndroidX BUCK to check if rules name are same or in sync.

**Comment by gengjiawen** [comment]
It's not just buck. Seems gradle need to change too. Windows is using gradle to build android RNTester.
```
* What went wrong:
Execution failed for task ':RNTester:android:app:lintVitalRelease'.
> Could not resolve all artifacts for configuration ':ReactAndroid:debugAndroidTestRuntimeClasspath'.
   > Could not find androidx.test:runner:1.0.2.
     Searched in the following locations:
       - file:/C:/android-sdk-windows/extras/m2repository/androidx/test/runner/1.0.2/runner-1.0.2.pom
       - file:/C:/android-sdk-windows/extras/m2repository/androidx/test/runner/1.0.2/runner-1.0.2.jar
       - file:/C:/android-sdk-windows/extras/google/m2repository/androidx/test/runner/1.0.2/runner-1.0.2.pom
       - file:/C:/android-sdk-windows/extras/google/m2repository/androidx/test/runner/1.0.2/runner-1.0.2.jar
       - file:/C:/android-sdk-windows/extras/android/m2repository/androidx/test/runner/1.0.2/runner-1.0.2.pom
       - file:/C:/android-sdk-windows/extras/android/m2repository/androidx/test/runner/1.0.2/runner-1.0.2.jar
       - file:/C:/Users/appveyor/.m2/repository/androidx/test/runner/1.0.2/runner-1.0.2.pom
       - file:/C:/Users/appveyor/.m2/repository/androidx/test/runner/1.0.2/runner-1.0.2.jar
       - https://dl.google.com/dl/android/maven2/androidx/test/runner/1.0.2/runner-1.0.2.pom
       - https://dl.google.com/dl/android/maven2/androidx/test/runner/1.0.2/runner-1.0.2.jar
       - https://jcenter.bintray.com/androidx/test/runner/1.0.2/runner-1.0.2.pom
       - https://jcenter.bintray.com/androidx/test/runner/1.0.2/runner-1.0.2.jar
       - file:/C:/projects/react-native/ReactAndroid/null/extras/m2repository/androidx/test/runner/1.0.2/runner-1.0.2.pom
       - file:/C:/projects/react-native/ReactAndroid/null/extras/m2repository/androidx/test/runner/1.0.2/runner-1.0.2.jar
     Required by:
         project :ReactAndroid
```

File: ReactAndroid/build.gradle [hunk 1]
```diff
@@ -337,8 +337,8 @@ dependencies {
     testImplementation("org.robolectric:robolectric:${ROBOLECTRIC_VERSION}")
 
     androidTestImplementation(fileTree(dir: "src/main/third-party/java/buck-android-support/", include: ["*.jar"]))
-    androidTestImplementation("com.android.support.test:runner:${ANDROID_SUPPORT_TEST_VERSION}")
-    androidTestImplementation("com.android.support.test:rules:${ANDROID_SUPPORT_TEST_VERSION}")
+    androidTestImplementation("androidx.test:runner:${ANDROID_SUPPORT_TEST_VERSION}")
```
Comment (gengjiawen) [inline ReactAndroid/build.gradle]: Change this to `1.1.0` ?
https://developer.android.com/jetpack/androidx/releases/test

**Comment by dulmandakh** [comment]
In the past, I tried to bump the Android Test Runner and it failed because
many tests were failing. So it might require some work @mdvacca

**Comment by mdvacca** [comment]
Landed as part of another PR

**Pull request closed without merging**